packages/swapper/src/swappers/osmosis/OsmosisSwapper.ts

```diff
@@ -375,7 +376,7 @@ export class OsmosisSwapper implements Swapper<ChainId> {
       }
 
       const ibcResponseAccount = await osmosisAdapter.getAccount(sellAddress)
-      const ibcAccountNumber = ibcResponseAccount.chainSpecific.accountNumber || '0'
+      const ibcAccountNumber = Number(ibcResponseAccount.chainSpecific.accountNumber)
```

Comment:
Not introduced by your PR by any means, but interestingly, while reading about cosmos-sdk BIP44 flavor, it seems like the actual variable component / terminology for account number derivation in CosmosHub is `address_index`, not `account`:

https://github.com/confio/cosmos-hd-key-derivation-spec#the-cosmos-hub-path

Looks like most cosmos-sdk blockchains use this path, however, we end up abstracting it as `accountNumber`, which from reading through chain-adapters / hdwallet implementations, is also what Keplr does.

At the end of the day, the derivation logic still ends up being correct with `addressNList` but we might want to be explicit about it in the future and use the `index` component as it should be